src/main/java/uk/gov/hmcts/reform/sandl/snlevents/actions/hearing/helpers/UserTransactionDataPreparerService.java

```diff
@@ -0,0 +1,56 @@
+package uk.gov.hmcts.reform.sandl.snlevents.actions.hearing.helpers;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.sandl.snlevents.exceptions.SnlRuntimeException;
+import uk.gov.hmcts.reform.sandl.snlevents.model.db.HearingPart;
+import uk.gov.hmcts.reform.sandl.snlevents.model.db.UserTransactionData;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+@Service
+public class UserTransactionDataPreparerService {
+    private List<UserTransactionData> userTransactionDataList = new ArrayList<>();
+
+    public List<UserTransactionData> getUserTransactionDataList() {
+        return userTransactionDataList;
+    }
+
+    public void prepareUserTransactionDataForUpdate(String entity, UUID entityId, String previousEntityString,
```

Comment:
Please rename this method, because now the name is inappropriate or leave the name method but remove action and counterAction parameters and hardcode them in function.

Reply:
Done